### ISSUE TITLE
Soap Logs 

### ISSUE THREAD

**Issue by res-io** [rubicon-project/node-google-dfp#31]
is there a way to get access to the soap logs somehow?

**Comment by AliUz** [comment]
I would also like to know this.

**Comment by grevory** [comment]
I have zero experience with SOAP. Does #34 cover the issue you have?

**Comment by serratedserenade** [comment]
Hey @grevory 

For the purposes of reporting bugs and/or DFP API difficulties, one must provide the SOAP logs over at the Google DFP API forums here https://groups.google.com/forum/#!forum/google-doubleclick-for-publishers-api

At the moment, using this node module, there doesn't seem to be an easy way (as far as I know) to grab the last request made (unless you open up access to the soap client as I've done in #34)

**Comment by Bennit** [comment]
I've made a PR to expose the last soap request body, as I needed the body to debug as well.
The internal soap client was not exposed and no reference was kept so I had to explictly expose it, and point to my forked solution.

I opened a PR for this feature: https://github.com/rubicon-project/node-google-dfp/pull/41
you use is as such:

```
dfpUser.getService('ReportService', function (err, service) {
  if (err) { /* log error */ )
  // build input ....
  const input = { ... }
  service.runReportjob(input, function (err, res) { 
    // ...
    console.log(service._lastSoapRequest());
  });
});
```

Two tips for using this library:
- check the official google dfp reference docs for specific types (xsi:type attribute is important!)
- the order of the elements also seems to matter so avoid using things like prettier on the places where you build your input objects for soap calls

**Comment by Bennit** [comment]
I guess this issue can now be closed now that the other PR doing the same thing was merged into master
https://github.com/rubicon-project/node-google-dfp/pull/34

**Comment by mvhirsch** [comment]
@grevory this can be closed, since (as @Bennit says) #34 was already merged.
